javaparser-core/src/main/javacc/java_1_8.jj

```diff
@@ -153,6 +153,16 @@ final class ASTParser {
 	    return ret;
     }
 
+    private ReferenceType juggleArrayTypes(Range range, Type type, int arrayCount, List<List<AnnotationExpr>> arrayAnnotations) {
+        if (arrayCount == 0) {
+            return new ReferenceType(range, type);
+        }
+        for (List<AnnotationExpr> annotations: arrayAnnotations) {
+            type = new ArrayType(range, type, annotations);
```

Comment:
Double-checked this and you're correct. I blame late night checkins. The funny thing is that I had it from right to left the first time around, but decided to change it for some reason.